### PR TITLE
Consolidate binary exponentiation files

### PR DESCRIPTION
**Pull request by tianyizheng02**
### Describe your change:

Consolidated implementations of binary (modular) exponentiation into just `binary_exponentiation.py`

* [ ] Add an algorithm?
* [ ] Fix a bug or typo in an existing algorithm?
* [ ] Add or change doctests? -- Note: Please avoid changing both code and tests in a single pull request.
* [x] Documentation change?

### Checklist:
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [ ] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [x] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [x] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [ ] All new algorithms include at least one URL that points to Wikipedia or another similar explanation.
* [ ] If this pull request resolves one or more open issues then the description above includes the issue number(s) with a [closing keyword](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue): "Fixes #ISSUE-NUMBER".

### CODE REVIEW

File: maths/binary_exponentiation.py [hunk 1]
```diff
@@ -1,48 +1,148 @@
...
+Repeat until b = 1 or b = 0
 
-def binary_exponentiation(a: int, n: int) -> int:
+For modular exponentiation, we use the fact that (a * b) % c = ((a % c) * (b % c)) % c
+"""
+
+
+def binary_exp_recursive(base: int, exponent: int) -> int:
     """
-    Compute a number raised by some quantity
```
Comment (cclauss) [inline maths/binary_exponentiation.py:9]: Why remove this?  `exp` is a bit cryptic in the function name and a line of documentation here is useful.

File: maths/binary_exponentiation.py [hunk 2]
```diff
@@ -1,48 +1,148 @@
...
-    >>> binary_exponentiation(-1, 3)
+    >>> binary_exp_recursive(3, 5)
```
Comment (cclauss) [inline maths/binary_exponentiation.py:19]: What about tests for a=big number andn=big number?

File: maths/binary_exponentiation.py [hunk 3]
```diff
@@ -1,48 +1,148 @@
...
-        POWER = int(input("Enter Power : ").strip())
-    except ValueError:
-        print("Invalid literal for integer")
+    BASE = int(input("Enter base: ").strip())
+    POWER = int(input("Enter power: ").strip())
 
-    RESULT = binary_exponentiation(BASE, POWER)
-    print(f"{BASE}^({POWER}) : {RESULT}")
+    RESULT = binary_exp_recursive(BASE, POWER)
+    print(f"{BASE}^({POWER}): {RESULT}")
```
Comment (cclauss) [inline maths/binary_exponentiation.py:148]: OPTIONAL: Adding a `timeit` benchmark would be a nice addition.  Visitors love to see the side-by-side bakeoff.

**Review by cclauss — APPROVED** [review]

File: maths/binary_exponentiation.py (hunk 2 above)
Comment (tianyizheng02) [reply maths/binary_exponentiation.py:21]: Can do, but the n=float case won't work because the algorithm can only compute integer powers

Comment (cclauss) [reply maths/binary_exponentiation.py:21]: I know it won’t work but I wanted to see how it fails.  Does it raise a ValueError?  Does it behave like `pow()` does.  Testing how things work if half the battle.  Watching them fail is just as cool.

**Pull request merged**